pkg/collector/corechecks/containers/containerd.go

```diff
@@ -0,0 +1,359 @@
+	cutil "github.com/DataDog/datadog-agent/pkg/util/containerd"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	containerdCheckName = "containerd"
+)
+
+// ContainerCheck grabs containerd metrics and events
+type ContainerdCheck struct {
+	core.CheckBase
+	hostname string
+	instance *ContainerdConfig
+	sub      *Subscriber
+}
+
+// ContainerdConfig contains the custom options and configurations set by the user.
+type ContainerdConfig struct {
+	Tags    []string `yaml:"tags"`
+	Filters []string `yaml:"filters"`
+}
+
+func init() {
+	corechecks.RegisterCheck(containerdCheckName, ContainerdFactory)
+}
+
+// ContainerdFactory is used to create register the check and initialize it.
+func ContainerdFactory() check.Check {
+	return &ContainerdCheck{
+		CheckBase: corechecks.NewCheckBase(containerdCheckName),
+		instance:  &ContainerdConfig{},
+		sub:       &Subscriber{},
+	}
+}
+
+// Parse is used to get the configuration set by the user
+func (co *ContainerdConfig) Parse(data []byte) error {
+	if err := yaml.Unmarshal(data, co); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Configure parses the check configuration and init the check
+func (c *ContainerdCheck) Configure(config, initConfig integration.Data) error {
+	var err error
+	if err = c.CommonConfigure(config); err != nil {
+		return err
+	}
+
+	c.hostname, err = util.GetHostname()
+	if err != nil {
+		return err
+	}
+
+	if err = c.instance.Parse(config); err != nil {
+		return err
+	}
+	c.sub.Filters = c.instance.Filters
+	return nil
+}
+
+// Run executes the check
+func (c *ContainerdCheck) Run() error {
+	sender, err := aggregator.GetSender(c.ID())
+	defer sender.Commit()
+	if err != nil {
+		return err
+	}
+	// As we do not rely on a singleton, we ensure connectivity every check run.
+	cu, errHealth := cutil.GetContainerdUtil()
+	if errHealth != nil {
+		log.Infof("Error ensuring connectivity with Containerd daemon %v", errHealth)
+		return errHealth
+	}
+	ns := cu.Namespace()
+
+	if c.sub == nil {
+		c.sub = CreateEventSubscriber("ContainerdCheck", ns, c.instance.Filters)
+	}
+
+	if !c.sub.IsRunning() {
+		// Keep track of the health of the Containerd socket
+		c.sub.CheckEvents(cu)
+	}
+	events := c.sub.Flush(time.Now().Unix())
+	// Process events
+	computeEvents(c.hostname, events, sender, c.instance.Tags)
+
+	nk := namespaces.WithNamespace(context.Background(), ns)
+	computeMetrics(sender, nk, cu, c.instance.Tags)
+
+	return nil
+}
+
+// compute events converts Containerd events into Datadog events
+func computeEvents(hostname string, events []ContainerdEvent, sender aggregator.Sender, userTags []string) {
+	for _, e := range events {
+		split := strings.Split(e.Topic, "/")
+		if len(split) != 3 {
+			// sanity checking the event, to avoid submitting
+			log.Tracef("Event topic %s does not have the expected format", e.Topic)
+			continue
+		}
+
+		output := metrics.Event{
+			Priority:       metrics.EventPriorityNormal,
+			Host:           hostname,
```

Comment:
The aggregator will add the agent's hostname automatically if the field is empty, I think we can remove this, and `c.hostname` to save a few lines